internal/pkg/nutanix/prismclient.go

```diff
@@ -0,0 +1,112 @@
+package nutanix
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	prismgoclient "github.com/nutanix-cloud-native/prism-go-client"
+	"github.com/nutanix-cloud-native/prism-go-client/utils"
+	v3 "github.com/nutanix-cloud-native/prism-go-client/v3"
+
+	"github.com/aws/eks-anywhere/pkg/providers/nutanix"
+)
+
+// PrismClient is a interface that provides useful functions for performing Prism operations.
+type PrismClient interface {
+	GetImageUUIDFromName(ctx context.Context, imageName string) (*string, error)
+	GetClusterUUIDFromName(ctx context.Context, clusterName string) (*string, error)
+	GetSubnetUUIDFromName(ctx context.Context, subnetName string) (*string, error)
+}
+
+type client struct {
+	v3.Client
```

Comment:
Do we want struct `client` to have an `is a` relationship with `v3.Client`?  This exposes all the behavior of the `v3.Client` from struct client. 

I think this would work for our use case
```code
type client struct {
	v3 v3.Client
}
```

Reply:
The client struct isn't exposed outside so it shouldn't matter. It just implements the `PrismClient` interface
So outside this package, you can only do 
```go
// initialize the client
prismClient := nutanix.NewPrismClient(endpoint, port, true)

// this is valid since you are calling the interface functions
prismClient.GetImageUUIDFromName()
prismClient.GetClusterUUIDFromName()
prismClient.GetSubnetUUIDFromName()

// this is not valid since the PrismClient interface doesn't have this function
// even though v3.Client has the ListSubnet() function
prismClient.V3.ListSubnet()
```

Reply:
Discussed offline. We can go with this approach.